generatedocs.py

```diff
@@ -0,0 +1,413 @@
+                baseClassMembers=baseClassMembers, 
+                members=groupedMembers,
+                examples=examples
+                )
+            with open(os.path.join(assetsOutputPath, assetComponent["name"]+'.md'), 'w') as f:
+                f.write(outputAssetComponent)
+
+    # Create index file
+    indexAssetComponentsTemplate = environment.get_template("indexAssetComponentsTemplate.txt")
+    outputIndex = indexAssetComponentsTemplate.render(assetCategories=assetCategories)
+    with open(assetsOutputPath + '/index.md', 'w') as f:
+        f.write(outputIndex)
+
+    # Print out missing assets
+    total = noOfFoundAssets + len(componentsMissingAssets)
+    percentFound = noOfFoundAssets / (total) * 100
+    print('\n\n')
+    print("Number of found asset examples (ignoring base classes):")
+    print(noOfFoundAssets, "of", total, "or", "%.1f" % percentFound, "%\n")
+    line = "-" * 80 
+    print(line)
+    print(len(componentsMissingAssets), "asset components are missing example files:")
+    print(line) 
+    print('\n'.join(componentsMissingAssets).join("\n\n\n"))
+
+
+################################################################################
+#                            CREATE SCRIPTING API                              #
+################################################################################
+
+def generateScriptingApi(environment, outputFolder, folderNameScripting, jsonLocation):
+    # Create target folder
+    scriptingOutputPath = os.path.join(outputFolder, folderNameScripting)
+    if not os.path.exists(scriptingOutputPath):
+        os.mkdir(scriptingOutputPath)
+
+    # Opening JSON file
+    f = open(os.path.join(jsonLocation, 'scriptingApi.json'))
+
+    # Convert JSON String to Python Dictionary
+    scriptingApi = json.load(f)
+
+    # Create the pages for the scripting libraries
+    scriptingApiTemplate = environment.get_template("scriptingApiTemplate.txt")
+    for library in scriptingApi:
+        # Go through all the functions in that library and print out a md file
+        library = parseDoxygenComments(library)
+        outputLibrary = scriptingApiTemplate.render(library=library)
+        with open(os.path.join(scriptingOutputPath, library["fullName"]+'.md'), 'w') as f:
+            f.write(outputLibrary)
+
+    # Create index file
+    indexScriptingTemplate = environment.get_template("indexScriptingTemplate.txt")
+    outputIndex = indexScriptingTemplate.render(libraries=scriptingApi)
+    with open(os.path.join(scriptingOutputPath, "index.md"), 'w') as f:
+        f.write(outputIndex)
+
+################################################################################
+#                         CREATE RENDERABLE OVERVIEW                           #
+################################################################################
+
+def generateRenderableOverview(environment, outputFolder, folderNameAssets, jsonLocation):
+    # Open JSON file
```

Comment:
The other functions start by creating a target folder. Why does this one not need to?

Reply:
Because this file is placed in the "base" folder in the output directory

![image](https://github.com/OpenSpace/OpenSpace-Docs/assets/24265984/4a882b76-5757-40cb-994b-571e9b7a4336)